src/components/autocomplete/js/autocompleteDirective.js

```diff
@@ -0,0 +1,58 @@
+(function () {
+  'use strict';
+  angular
+      .module('material.components.autocomplete')
+      .directive('mdAutocomplete', MdAutocomplete);
+
+  function MdAutocomplete () {
+    return {
+      template: '\
+        <md-autocomplete-wrap role="listbox">\
+          <input type="text"\
+              ng-model="searchText"\
+              ng-keydown="$mdAutocompleteCtrl.keydown($event)"\
+              placeholder="{{placeholder}}"\
+              aria-label="{{placeholder}}"\
+              aria-autocomplete="list"\
+              aria-haspopup="true"\
+              aria-activedescendant=""\
+              aria-expanded="{{!$mdAutocompleteCtrl.hidden}}"\
+              ng-keydown="$mdAutocompleteCtrl.onKeydown($event)"/>\
+          <button\
+              type="button"\
+              ng-if="searchText"\
+              ng-click="$mdAutocompleteCtrl.clear()">\
```

Comment:
When this button has focus and you clear the field, focus is lost. We should handle it by sending focus back to the input.